sdk/metrics/src/ViewProxy.cpp

```diff
@@ -30,72 +30,74 @@ libmexclass::proxy::MakeResult ViewProxy::make(const libmexclass::proxy::Functio
     auto unit = unit_mda[0];
 
     matlab::data::TypedArray<int> aggregation_type_mda = constructor_arguments[9];
-    auto aggregation_type =  static_cast<metrics_sdk::AggregationType>(static_cast<int>(aggregation_type_mda[0]));
+    metrics_sdk::AggregationType aggregation_type;
+    if(static_cast<int>(aggregation_type_mda[0]) == -1){
+        aggregation_type =  metrics_sdk::AggregationType::kDefault;
+    }else{
+        aggregation_type =  static_cast<metrics_sdk::AggregationType>(static_cast<int>(aggregation_type_mda[0]));
+    }
 
     std::shared_ptr<metrics_sdk::HistogramAggregationConfig> aggregation_config = std::shared_ptr<metrics_sdk::HistogramAggregationConfig>(new metrics_sdk::HistogramAggregationConfig());
     if(aggregation_type == metrics_sdk::AggregationType::kHistogram){
         matlab::data::TypedArray<double> histogramBinEdges_mda = constructor_arguments[10];
         std::vector<double> histogramBinEdges;
-        for (auto h : histogramBinEdges_mda) {
-            histogramBinEdges.push_back(h);
+        if(histogramBinEdges_mda.getNumberOfElements() > 0){
```

Comment:
Is this if statement necessary? I would think if histogramBinEdges_mda is empty, the for loop will not run.